docs/basic-features/environment-variables.md

```diff
@@ -46,8 +46,7 @@ export async function getStaticProps() {
 }
 ```
 
-> **Note**: In order to keep server-only secrets safe, Next.js replaces `process.env.*` with the correct values
-> at build time. This means that `process.env` is not a standard JavaScript object, so you’re not able to
+> **Note**: In order to keep server-only secrets safe, environment variables are evaluated at build time, so only environment variables *actually* used will be included. This means that `process.env` is not a standard JavaScript object, so you’re not able to
```

Comment:
This is less clear because "evaluated" is not the same as "replaced".

Typically eval allows dynamic usage but this is pure static analysis. The syntax is very specific.

Reply:
You are right. Any suggestion on how we can make it more clear? :thinking:

Reply:
I think this entire page needs to be reworked but we can ship this for now